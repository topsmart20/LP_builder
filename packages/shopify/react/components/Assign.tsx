import React from 'react';
import { Builder, BuilderStore } from '@builder.io/react';

interface AssignBlockProps {
  expression?: string;
  builderState?: BuilderStore;
}

export class AssignBlock extends React.Component<AssignBlockProps> {
  constructor(props: AssignBlockProps) {
    super(props);

    const { expression, builderState } = props;

    if (expression && builderState) {
      builderState.state.shopify?.liquid?.assign(expression);
    }
  }
}

Builder.registerComponent(AssignBlock, {
  name: 'Shopify:Assign',
  hideFromInsertMenu: true,
  inputs: [
    {
      name: 'expression',
      type: 'javascript',
      // TODO: type: 'code', language: 'javascript'
    },
  ],
});
