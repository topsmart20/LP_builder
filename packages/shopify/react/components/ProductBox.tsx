/** @jsx jsx */
import { jsx, css, keyframes } from '@emotion/core';

import React, { useEffect, useState } from 'react';
import { Builder, BuilderElement } from '@builder.io/sdk';
import { BuilderStore, BuilderStoreContext, BuilderBlocks } from '@builder.io/react';
import { modifyProduct } from '../functions/modify-product';

interface ProductBoxProps {
  product?: string | number;
  /** @todo implement this */
  variant?: string | number;
  builderState?: BuilderStore;
  builderBlock?: BuilderElement;
}

const spin = keyframes`
  0% {
    transform:rotate(0)
  }
  100% {
    transform:rotate(360deg)
  }
`;

function LoadingSpinner() {
  return (
    <div
      css={css`
        width: 4em;
        height: 4em;
        border-radius: 50%;
        position: relative;
        margin: 6rem auto;
        font-size: 1rem;
        text-indent: -9999em;
        border-top: 0.2em solid rgba(131, 132, 137, 0.2);
        border-right: 0.2em solid rgba(131, 132, 137, 0.2);
        border-bottom: 0.2em solid rgba(131, 132, 137, 0.2);
        border-left: 0.2em solid #454749;
        transform: translateZ(0);
        animation: 1.1s linear infinite ${spin};
      `}
    />
  );
}

export function ProductBox(props: ProductBoxProps) {
  const [productInfo, setProductInfo] = useState(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    updateProduct();
  }, [props.product]);

  function updateProduct() {
    setLoading(true);
    const { context } = props.builderState!;
    let productId = props.product || '';

    if (productId) {
      fetch(`https://builder.io/api/v1/shopify/products/${productId}.json?apiKey=${context.apiKey}`)
        .then(res => res.json())
        .then(data => {
          const { product } = data;
          const modifiedProduct = modifyProduct(product);
          setProductInfo(modifiedProduct);
          setLoading(false);
        })
        .catch(err => {
          console.error('Error fetching Shopify product', err);
          setLoading(false);
        });
    } else {
      setProductInfo(null);
    }
  }

  return loading && !productInfo ? (
    <LoadingSpinner />
  ) : (
    <BuilderStoreContext.Consumer>
      {store => (
        <BuilderStoreContext.Provider
          value={{
            ...store,
            state: {
              ...store.state,
              // TODO: communicate up this up to DataEditor to populate bindings dropdowns (how?)
              productInfo,
            },
          }}
        >
          <BuilderBlocks
            child
            parentElementId={props.builderBlock?.id}
            blocks={props.builderBlock?.children}
          />
        </BuilderStoreContext.Provider>
      )}
    </BuilderStoreContext.Consumer>
  );
}

Builder.registerComponent(ProductBox, {
  name: 'Shopify:ProductBox',
  canHaveChildren: true,
  hideFromInsertMenu: true,
  inputs: [
    {
      name: 'product',
      type: 'ShopifyProduct',
    },
  ],
});
