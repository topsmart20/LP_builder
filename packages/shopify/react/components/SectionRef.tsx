import React, { useState, useEffect } from 'react';
import { Builder } from '@builder.io/react';

interface SectionRefProps {
  class?: string;
  section?: string;
}
const cache: { [key: string]: string } = {};

export const SectionRef = (props: SectionRefProps) => {
  const [html, setHtml] = useState<string | null>(null);
  const sectionName = props.section?.split('/')[1].split('.')[0]!;

  useEffect(() => {
    if (cache[sectionName]) {
      return;
    }
    // Convert `sections/foo.liquid` -> `foo`
    if (!sectionName) {
      return;
    }
    if (cache[sectionName]) {
      setHtml(cache[sectionName]);
      return;
    }
    // This may look strange, but it is how Shopify's section rendering API
    // works https://shopify.dev/docs/themes/sections/section-rendering-api
    fetch(`${location.search ? '&' : '?'}section_id=${sectionName}`)
      .then(res => res.text())
      .then(text => {
        cache[sectionName] = text;
        setHtml(text);
      });
  }, [sectionName]);

  return (
    <div
      className={`shopify-section ${props.class}`}
      dangerouslySetInnerHTML={{ __html: html || cache[sectionName] || '' }}
    />
  );
};

Builder.registerComponent(SectionRef, {
  name: 'Shopify:SectionRef',
  hideFromInsertMenu: true,
});
