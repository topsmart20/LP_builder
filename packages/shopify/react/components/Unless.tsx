import * as React from 'react';
import builder, {
  Builder,
  BuilderStore,
  onChange,
  withBuilder,
  BuilderElement,
  stringToFunction,
  BuilderBlocks,
} from '@builder.io/react';

interface UnlessBlockProps {
  builderState?: BuilderStore;
  builderBlock?: BuilderElement;
  unlessBlocks?: BuilderElement[];
  elseBlocks?: BuilderElement[];
  expression?: string;
}

export class UnlessBlock extends React.Component<UnlessBlockProps> {
  private getMatchingBlocks() {
    const { expression, unlessBlocks, elseBlocks } = this.props;
    if (!expression) {
      return unlessBlocks;
    }
    const fn = stringToFunction(expression);
    const result = fn(
      this.props.builderState && this.props.builderState.state,
      null,
      this.props.builderBlock,
      builder,
      null,
      this.props.builderState!.update,
      Builder,
      this.props.builderState?.context
    );
    return result ? elseBlocks : unlessBlocks;
  }

  render() {
    const { unlessBlocks } = this.props;
    const blocks = this.getMatchingBlocks();
    if (!blocks) {
      return null;
    }
    const isUnless = blocks === unlessBlocks;

    return (
      <BuilderBlocks
        child
        parentElementId={this.props.builderBlock && this.props.builderBlock.id}
        blocks={blocks}
        dataPath={`component.options.${isUnless ? 'unlessBlocks' : 'elseBlocks'}`}
      />
    );
  }
}

const defaultBlock = {
  '@type': '@builder.io/sdk:Element',
  responsiveStyles: {
    large: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'stretch',
      flexShrink: '0',
      position: 'relative',
      marginTop: '30px',
      textAlign: 'center',
      lineHeight: 'normal',
      height: 'auto',
    },
  },
  component: {
    name: 'Text',
    options: {
      text: '<p>Enter some text...</p>',
    },
  },
};

withBuilder(UnlessBlock, {
  name: 'Shopify:Unless',
  hideFromInsertMenu: true,
  noWrap: true,
  inputs: [
    {
      name: 'expression',
      type: 'javascript',
    },
    {
      name: 'unlessBlocks',
      type: 'array',
      hideFromUI: true,
      defaultValue: [defaultBlock],
    },
    {
      name: 'elseBlocks',
      type: 'array',
      hideFromUI: true,
      defaultValue: [defaultBlock],
    },
  ],
});
