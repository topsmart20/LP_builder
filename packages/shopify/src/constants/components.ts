import { BuilderElement } from "@builder.io/sdk";
import { Options } from "../interfaces/options";

export interface ComponentInfo {
  component: (block: BuilderElement, options: Options, attributes?: string) => string;
  noWrap?: boolean;
  name: string;
}

export function component(info: ComponentInfo) {
  components[info.name] = info;
  return info.component;
}

export function getComponentInfo(name: string) {
  return components[name];
}

export function getComponent(name: string) {
  const info = getComponentInfo(name)
  return info && info.component;
}

export const components: { [key: string]: ComponentInfo | undefined } = {};
