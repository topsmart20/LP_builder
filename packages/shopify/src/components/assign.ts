import { BuilderElement } from '@builder.io/sdk';
import { Options } from '../interfaces/options';
import { component } from '../constants/components';

export const Assign = component({
  name: 'Shopify:Assign',
  noWrap: true,
  component: (block: BuilderElement, renderOptions: Options) => {
    const { options } = block.component!;
    const content = options.expression;

    return `{% assign ${content} %}`;
  },
});
