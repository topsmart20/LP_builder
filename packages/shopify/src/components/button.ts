import { component } from '../constants/components';

export const Button = component({
  name: 'Core:Button',
  noWrap: true,
  component: (block, renderOptions, attributes) => {
    const { options } = block.component!;
    const tag = options.link ? 'a' : 'span';

    return `
      <${tag}${options.openLinkInNewTab ? ' target="_blank"' : ''} ${attributes || ''}>
        ${options.text || ''}
      </${tag}>
    `;
  },
});
