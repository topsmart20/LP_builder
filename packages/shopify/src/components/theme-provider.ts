import { BuilderElement } from '@builder.io/sdk';
import { Options } from '../interfaces/options';
import { blockToLiquid } from '../functions/block-to-liquid';
import { component } from '../constants/components';

const postProcessThemeLiquid = (liquid: string) => {
  return liquid.replace(/([^a-z_\.])settings\s*\.\s*/gi, `$1_theme_settings.`);
};
export const ThemeProvider = component({
  name: 'Shopify:ThemeProvider',
  noWrap: true,
  component: (block: BuilderElement, renderOptions: Options) => {
    const liquid = `
      ${
        block.children
          ? block.children.map(child => blockToLiquid(child, renderOptions)).join('\n')
          : ''
      }
  `;
    return postProcessThemeLiquid(liquid);
  },
});
