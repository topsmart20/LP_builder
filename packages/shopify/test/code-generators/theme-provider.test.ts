import { ThemeProvider } from '../../src/components/theme-provider';
import { el } from '../modules/helpers';
import * as simplePage from '../pages/compare/simple.json';
import { BuilderElement } from '@builder.io/sdk';

const mockState = {
  settings: {
    key: 'value',
    boolean_key: true,
    number_key: 1,
  },
};

const getThemeProviderCode = (options: any, children: BuilderElement[] = []): string =>
  ThemeProvider(
    el({
      children,
      component: {
        options,
        name: 'Shopify:ThemeProvider',
      },
    }),
    {}
  );

describe('ThemProvider code generator', () => {
  test('no values for attributes', () => {
    const code = getThemeProviderCode({});
    expect(code).toMatchSnapshot();
  });

  test('with children', () => {
    const children = simplePage.data.blocks as BuilderElement[];
    const code = getThemeProviderCode({ state: mockState }, children);
    expect(code).toMatchSnapshot();
  });
});
