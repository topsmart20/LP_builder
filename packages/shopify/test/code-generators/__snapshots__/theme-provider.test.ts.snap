// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemProvider code generator no values for attributes 1`] = `
"
      
  "
`;

exports[`ThemProvider code generator with children 1`] = `
"
      
    <style>.builder-block.builder-9c660eec3a084703acfd457a5f46bec0 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: relative;
  flex-shrink: 0;
  box-sizing: border-box;
  margin-top: 20px;}</style>
    
    
    
    
      <div  builder-id=\\"builder-9c660eec3a084703acfd457a5f46bec0\\" class=\\"builder-block builder-9c660eec3a084703acfd457a5f46bec0\\">
        
    <style>
      .builder-columns {
        display: flex;
      }

      .builder-column > .builder-blocks {
        flex-grow: 1;
      }
      
      .builder-9c660eec3a084703acfd457a5f46bec0 > .builder-columns > .builder-column:nth-child(1) {
        width: calc(50% - 10px);
        margin-left: 0px;
      }

      .builder-9c660eec3a084703acfd457a5f46bec0 > .builder-columns > .builder-column:nth-child(2) {
        width: calc(50% - 10px);
        margin-left: 20px;
      }
      </style>
      <div class=\\"builder-columns\\">
        <div
            class=\\"builder-column\\"
            style=\\"line-height: normal;display: flex;flex-direction: column;align-items: stretch;\\"
            >
            <div
              class=\\"builder-blocks\\"
              builder-type=\\"blocks\\">
              
    <style>.builder-block.builder-b6a0a9174cd94d48b555e03f28ebb536 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex-shrink: 0;
  position: relative;
  margin-top: 30px;
  text-align: center;
  line-height: normal;
  height: auto;}</style>
    
    
    
    
      <div  builder-id=\\"builder-b6a0a9174cd94d48b555e03f28ebb536\\" class=\\"builder-block builder-b6a0a9174cd94d48b555e03f28ebb536\\">
        
      <picture>
        <source srcset=\\"https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=100 100w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=200 200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=400 400w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=800 800w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=1200 1200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=1600 1600w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=2000 2000w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d\\" type=\\"image/webp\\" />
        <img
          src=\\"https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d\\"
          srcset=\\"https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=100 100w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=200 200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=400 400w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=800 800w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=1200 1200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=1600 1600w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=2000 2000w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d\\"
          sizes=\\"100vw\\"
          style=\\"object-fit: cover;object-position: center;position: absolute;height: 100%;width: 100%;top: 0;left: 0;\\" />
      </picture>
      <div
          class=\\"builder-image-sizer\\"
          style=\\"width: 100%;padding-top: 70.04048582995948%;pointer-events: none;font-size: 0;\\"></div>
        
  
      </div>
    
    
    
  

    <style>.builder-block.builder-6dc1308248be46e18b58ad4c623916d8 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex-shrink: 0;
  position: relative;
  margin-top: 30px;
  text-align: center;
  line-height: normal;
  height: auto;}</style>
    
    
    
    
      <div  builder-id=\\"builder-6dc1308248be46e18b58ad4c623916d8\\" class=\\"builder-block builder-6dc1308248be46e18b58ad4c623916d8\\">
        
      <style>
        .builder-text p:first-of-type, .builder-paragraph:first-of-type { margin: 0 } .builder-text > p, .builder-paragraph { color: inherit; line-height: inherit; letter-spacing: inherit; font-weight: inherit; font-size: inherit; text-align: inherit; font-family: inherit; }
      </style>
      <span class=\\"builder-text\\">
        <p>Enter some text...</p>
      </span>
    
      </div>
    
    
    
  
            </div>
          </div>
<div
            class=\\"builder-column\\"
            style=\\"line-height: normal;display: flex;flex-direction: column;align-items: stretch;\\"
            >
            <div
              class=\\"builder-blocks\\"
              builder-type=\\"blocks\\">
              
    <style>.builder-block.builder-a8fe67f4bed845a38528c106f08d9b45 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex-shrink: 0;
  position: relative;
  margin-top: 30px;
  text-align: center;
  line-height: normal;
  height: auto;}</style>
    
    
    
    
      <div  builder-id=\\"builder-a8fe67f4bed845a38528c106f08d9b45\\" class=\\"builder-block builder-a8fe67f4bed845a38528c106f08d9b45\\">
        
      <picture>
        <source srcset=\\"https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=100 100w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=200 200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=400 400w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=800 800w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=1200 1200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=1600 1600w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?format=webp&width=2000 2000w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d\\" type=\\"image/webp\\" />
        <img
          src=\\"https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d\\"
          srcset=\\"https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=100 100w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=200 200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=400 400w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=800 800w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=1200 1200w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=1600 1600w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d?width=2000 2000w, https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d\\"
          sizes=\\"100vw\\"
          style=\\"object-fit: cover;object-position: center;position: absolute;height: 100%;width: 100%;top: 0;left: 0;\\" />
      </picture>
      <div
          class=\\"builder-image-sizer\\"
          style=\\"width: 100%;padding-top: 70.04048582995948%;pointer-events: none;font-size: 0;\\"></div>
        
  
      </div>
    
    
    
  

    <style>.builder-block.builder-25ba5db0c872489898ea10f2816c08b6 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex-shrink: 0;
  position: relative;
  margin-top: 30px;
  text-align: center;
  line-height: normal;
  height: auto;}</style>
    {% for yo_item in thing.list.yo %}
    
    
    
      <a  href=\\"{{ bazbaz }}\\" builder-id=\\"builder-25ba5db0c872489898ea10f2816c08b6\\" class=\\"builder-block builder-25ba5db0c872489898ea10f2816c08b6\\" style=\\"display: {{ foo.bar }};\\">
        
      <style>
        .builder-text p:first-of-type, .builder-paragraph:first-of-type { margin: 0 } .builder-text > p, .builder-paragraph { color: inherit; line-height: inherit; letter-spacing: inherit; font-weight: inherit; font-size: inherit; text-align: inherit; font-family: inherit; }
      </style>
      <span class=\\"builder-text\\">
        {{ foobar }}
      </span>
    
      </a>
    
    
    {% endfor %}
  
            </div>
          </div>
        
          <style>
        @media (max-width: 999px) {
          .builder-9c660eec3a084703acfd457a5f46bec0 > .builder-columns {
            flex-direction: column;
            align-items: stretch;
          }

          .builder-9c660eec3a084703acfd457a5f46bec0 > .builder-columns > .builder-column:nth-child(n) {
            width: 100%;
            margin-left: 0;
          }
        }
          </style>
        
      </div>
  
      </div>
    
    
    
  
  "
`;
